Funcky/Extensions/StringExtensions/SplitLines.cs

```diff
@@ -0,0 +1,55 @@
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using Funcky.Monads;
+
+namespace Funcky.Extensions
+{
+    public static partial class StringExtensions
+    {
+        private delegate int GetLength(int index, bool hasCarriageReturn);
+
+        /// <summary>
+        /// Splits a string into individual lines lazily, by any new line (CR, LF, CRLF).
```

Comment:
CR by itself is not a line terminator

Reply:
C# documentation says: 

A line is defined as a sequence of characters followed by a line feed ("\n"), a carriage return ("\r"), a carriage return immediately followed by a line feed ("\r\n"), or the end-of-stream marker.

this is how it works for:
StringReader, StreamReader, File.ReadLines and File.ReadAllLines - to change this would awkward.